mllib/src/main/scala/org/apache/spark/ml/Predictor.scala

```diff
@@ -62,6 +62,40 @@ private[ml] trait PredictorParams extends Params
     }
     SchemaUtils.appendColumn(schema, $(predictionCol), DoubleType)
   }
+
+  /**
+   * Extract [[labelCol]], weightCol(if any) and [[featuresCol]] from the given dataset,
```

Comment:
I place it in `PredictorParam` so that methods like `GBTModel.evaluateEachIteration` can reuse it in the future.